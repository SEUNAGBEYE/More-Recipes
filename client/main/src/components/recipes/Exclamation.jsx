import React from 'react';
import PropTypes from 'prop-types';

/**
 * @export
 * @param {Object} props
 *
 * @returns  {jsx} JSX
 */
const Exclamation = (props) => (
  <div className="text-center exclamation-container">
    {props.children}
    <div className="fa fa-exclamation-circle center-exclamation" />
  </div>
);

const propTypes = {
  children: PropTypes.object
};

Exclamation.propTypes = propTypes;

export default Exclamation;
