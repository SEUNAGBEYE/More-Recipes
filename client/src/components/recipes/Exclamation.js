import React from 'react';

export default function(){
  return(
      <div className='text-center exclamation-container'>
        <p className='text-muted'>Sorry you haven't added any recipe yet, please add to get started</p>
        <div className='fa fa-exclamation-circle center-exclamation'></div>
      </div>
  )
}