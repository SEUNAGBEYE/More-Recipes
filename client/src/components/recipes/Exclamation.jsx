import React from 'react';

export default function(props){
  return(
      <div className='text-center exclamation-container'>
        {props.children}
        <div className='fa fa-exclamation-circle center-exclamation'></div>
      </div>
  )
}