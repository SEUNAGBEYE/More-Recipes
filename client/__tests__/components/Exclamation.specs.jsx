
import React from 'react';
import { shallow } from 'enzyme';
import Exclamation from '../../main/src/components/recipes/Exclamation';

describe('# Exclamation', () => {
  const wrapper = shallow(<Exclamation/>);
  it('should render successfully', (done) => {
    expect(wrapper).toBeDefined();
    expect(wrapper.getElement().type).toBe('div');
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
