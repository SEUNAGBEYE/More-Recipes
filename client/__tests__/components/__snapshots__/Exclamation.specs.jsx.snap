// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Exclamation should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Exclamation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <div
          className="fa fa-exclamation-circle center-exclamation"
        />,
      ],
      "className": "text-center exclamation-container",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fa fa-exclamation-circle center-exclamation",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <div
            className="fa fa-exclamation-circle center-exclamation"
          />,
        ],
        "className": "text-center exclamation-container",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-exclamation-circle center-exclamation",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
